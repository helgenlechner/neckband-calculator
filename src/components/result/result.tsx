import React, { PureComponent } from 'react';

interface Props {
  length: number | undefined;
}

export class Result extends PureComponent<Props> {
  render () {
    const { length } = this.props;

    if (length === undefined) {
      return null;
    }

    return (
      <p>The neck band needs to be {length.toFixed(1)}cm long, including seam allowances.</p>
    );
  }
}
