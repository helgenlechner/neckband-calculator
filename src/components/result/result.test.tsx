import React from 'react';
import renderer from 'react-test-renderer';
import { Result } from './result';

describe('Result Component', () => {
    it('renders null if the length is not defined', () => {
        const tree = renderer
            .create(<Result />)
            .toJSON();

        expect(tree).toMatchSnapshot();
    });

    it('renders a <p> with a sentence including the length rounded to one digit after the period', () => {
        const tree = renderer
            .create(<Result length={Math.PI} />)
            .toJSON();

        expect(tree).toMatchSnapshot();
    });
});
