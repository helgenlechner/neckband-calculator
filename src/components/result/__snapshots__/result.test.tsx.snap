// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Result Component renders a <p> with a sentence including the length rounded to one digit after the period 1`] = `
<p
  data-test-id="result"
>
  The neck band needs to be 
  3.1
  cm long, including seam allowances.
</p>
`;

exports[`Result Component renders null if the length is not defined 1`] = `null`;
